transform/coroutines.go

```diff
@@ -5,11 +5,27 @@ package transform
 
 import (
 	"errors"
-	"github.com/tinygo-org/tinygo/compiler/llvmutil"
 	"strconv"
+
+	"github.com/tinygo-org/tinygo/compiler/llvmutil"
 	"tinygo.org/x/go-llvm"
 )
 
+// These functions need to be preserved in the IR until after the coroutines
+// pass has run.
+var coroFunctionsUsedInTransforms = []string{
+	"internal/task.start",
+	"internal/task.Pause",
+	"internal/task.fake",
+	"internal/task.Current",
+	"internal/task.createTask",
+	"(*internal/task.Task).setState",
+	"(*internal/task.Task).returnTo",
+	"(*internal/task.Task).returnCurrent",
+	"(*internal/task.Task).setReturnPtr",
+	"(*internal/task.Task).getReturnPtr",
+}
```

Comment:
These should be put near their use - maybe right after `functionsUsedInTransforms`.